### ISSUE TITLE
Support GitLab

### ISSUE THREAD

**Issue by TheCodeDestroyer** [himynameisdave/git-labelmaker#48]
Just a question(type), I see that most of the code refers to GitHub,
but I was wondering if you are thinking of maybe supporting [GitLab](http://doc.gitlab.com/ce/api/labels.html) as well?

P.S.: The only decent NodeJS Gitlab API wrapper I found is [this](https://github.com/node-gitlab/node-gitlab) one, but doesn't support labels from what I can see...

**Comment by himynameisdave** [comment]
**Short answer:** yes, eventually...

It was kindly [pointed out to me on Reddit](https://www.reddit.com/r/webdev/comments/470pfx/sane_github_labels/d09db51) that naming this project `git-labelmaker` when it actually only applies to GitHub was kind of a goof. So as a result I feel like in the future we should support GitLab and BitBucket, or maybe just rename this tool `github-labelmaker` and create a separate tool for GitLab.

Now in terms of a timeframe, I'm not going to lie there are other features we would like to add here first.

Thanks for your question!

**Comment by Jameskmonger** [comment]
I initially added this to the `0.9.0` milestone, but I haven't used GitLab before. I'd like to take a bit more of a look into their API.

**Comment by himynameisdave** [comment]
Yeah this isn't a priority for `0.9.0`, I'm thinking this is a significant change that would be better for a `1.x` release

**Comment by TheCodeDestroyer** [comment]
Since your code tightly couples to GitHub API, maybe it wouldn't be a bad idea to rename this to "github-labelmaker" and make the other two as separate projects. You probably don't want to get into the battle where one change breaks the other two and vice-versa.

**Comment by Jameskmonger** [comment]
I think we could potentially have this as a project which supports labelling for the different platforms, we can just let the user select which service they want to use.

**Comment by dxlr8r** [comment]
Would love gitlab support as well :)
